### ISSUE TITLE
BUG #13475 still persistant[color calibration] _(AI) detect from image surfaces_ does not detect

### ISSUE THREAD

**Issue by AxelG-DE** [darktable-org/darktable#15375]
### Describe the bug

What I have described in #13475 still happens to me.

Hard to nail down is the following observation:

- I tried several times with files from Olympus OMD EM-5 MK II and it seemed to work
- then I went to a set of Nikon D750 and the bug is immediately there
- back to previous set, now also fails on Olympus' files (means 5002k every time, in rare occasions 5001k)

Indeed I test on total different image appearances.

I quickly alligned with @jenshannoschwalm on IRC and hence I reopen here

I wonder whether @ralfbrown now is free from this bug or also still sees it.

would I need to record a screen-cast in order to prove? 

### Steps to reproduce

see #13475

### Expected behavior

there should be a reliably different colorcalibration "reaction" based on the scene

### Where did you install darktable from?

self compiled

### darktable version

4.5.0+804~gf0c26f6cc1

### What OS are you using?

Linux

### What is the version of your OS?

gentoo

### Describe your system?

see additional context

### Are you using OpenCL GPU in darktable?

Yes

### If yes, what is the GPU card and driver?

see below

### Please provide additional context if applicable. You can attach files too, but might need to rename to .txt or .zip

* darktable version : 4.5.0+804~gf0c26f6cc1
* OS                : Linux - kernel 6.5.5-gentoo
* Distro            : Gentoo Base System release 2.14
* Processor         : Intel(R) Core(TM) i9-9900K CPU @ 3.60GHz
* Memory            : 32 GB (4 x 8 GB) + 5GB Swap
* Graphics card0    : NVIDIA GeForce GTX 1060 6GB
* Graphics card1    : NVIDIA GeForce RTX 2070 SUPER
* Graphics driver   : nvidia-drivers-535.104.05
* OpenCL installed  : Yes (opencl-headers-2023.04.17)
* OpenCL activated  : Yes
* Xorg              : xorg-server-21.1.8-r2
* Desktop           : KDE 5
* GTK+              : gtk+-3.24.38
* gcc               : 13.2.1_p20230826
* cflags            : CMAKE_FLAGS="-march=native-O2-mtune=native-pipe"
* CMAKE_BUILD_TYPE  : "Release"

**Comment by jenshannoschwalm** [comment]
I can confirm the strong tendency to 5002°, some images don't do so, all images have something changed, i also doubt the intelligence of the "AI" model.

I can __not__ confirm any "instability" like getting different results for different runs on the same image without changing some module's settings that affects input data for "color calibration". That would certainly be a bug - exactly that bug i mentioned and thought to be fixed.

As i can't reproduce yet it would be helping to rule out pixelpipe cache issues. So these questions:
1. can you reproduce your issue with `--disable-pipecache`? That would completely rule out such problems.
2. it would be perfect to get a fresh screencast with time comments "when do you see what concidered to be wrong", a log with `-d pipe` would help too.

**Comment by AxelG-DE** [comment]
I did actually discard history yesterday, when I started really digging into it, but also I do have a lot of auto-apply thingies, so when opening the pic in Darkroom, it is not unaltered anymore...

Yes, I just tested, with --disable-pipecache the issue persists

I will prepare the screencast tomorrow (I need some time to collect a set w/o family members or other ppl in it and today is booked).

**Comment by AxelG-DE** [comment]
Side comment: 
Why I am interested in this so called AI thingie? I do not agree so much with Aurélien, where he said, fetch a sample (e.g. gray or even gray-card) which is irrediated by the (main) light source. If I do so, I eliminate the e.g. warmth of a sunset (yes can be colour gradet afterwards). I tend to fetch a sample of gray or black which is in the shade. Might be wrong from scientific point, but this is, how I like my photos. Sometimes, when the sun is behind the photographer, it is difficult to find a shaded sample in gray. The standard color picker does not satisfy me in all cases either...

sorry for this off-topic

I donno why, I have several shots from this scene and here I have no 5002° result in none of them (.ORF)
![grafik](https://github.com/darktable-org/darktable/assets/49004103/a280e355-defe-434b-9970-e814daf28ba7)

**Comment by AxelG-DE** [comment]
- sorry, the video is so rushy. I thought it could make the vid smaler, but finally no...
My observations
- It is really hard to say, which is right and which is wrong. Some might be debatable
- When there is a "potentially correct" result, I notice a color-flicker (e.g. 00:45, 02:14, 02:18)
- I'd consider all results with 5002° as potentially incorrect
- The behaviour is independent from initial values for illuminant or temperature respectively hue
- as I recorded this video two times, there is one strange issue. Believe it or not, there is a set of above shown staircase. One of those DID show a 5002 in the first run, but NOT in the second turn


The output of -d pipe of the session recorded below
[5002-issue__-d_pipe.txt.zip](https://github.com/darktable-org/darktable/files/12838117/5002-issue__-d_pipe.txt.zip)

Please rename to *.rar instead of *.zip
[ColorCal-15375_20231007-2023-10-07_17.12.32.mp4.part1.zip](https://github.com/darktable-org/darktable/files/12838133/ColorCal-15375_20231007-2023-10-07_17.12.32.mp4.part1.zip)
[ColorCal-15375_20231007-2023-10-07_17.12.32.mp4.part2.zip](https://github.com/darktable-org/darktable/files/12838134/ColorCal-15375_20231007-2023-10-07_17.12.32.mp4.part2.zip)

**Comment by ralfbrown** [comment]
I just tried a bunch of images in my folder of PlayRaw pics and still get 5001-5003K on nearly all.  A couple of sunset pics had 499xK, and a third around 3600K (completely orange sky and silhouetted foreground).  The only other pic that didn't get 4990-5003K was a night-time campfire at 2538K.

That's pretty much what my experience with the surface-AI option has been ever since it was introduced - the rare cases where it doesn't result in 5002K (+/-) aren't worth the time spent trying it and then having to set thing manually anyway.

**Comment by jenshannoschwalm** [comment]
So what to make out of this. A bug for me would be inconsistent results. What I understand here is just a non-satisfactory algo, right?

**Comment by AxelG-DE** [comment]
I wouldn't say a bug is only inconsistency. A function, which (potentially) not working as intended  (assume the slider for exposure would slide and change values on GUI but not do anything to the picture), I also would consider this a bug :)

**Comment by ralfbrown** [comment]
Well, it does change the sliders (and thus colors) for me, but nearly always to a default value that doesn't necessarily have anythiung to do with the image.

**Comment by jenshannoschwalm** [comment]
@AxelG-DE can you reproduce with pipe ache disabled? In your log it's enabled.

**Comment by AxelG-DE** [comment]
@jenshannoschwalm I confirmed already prior this screencast, above, that yes, with pipe-cache disabled I can reproduce :)

**Comment by jenshannoschwalm** [comment]
I really don't know if this is just an "algo sucks" problem...

**Comment by AxelG-DE** [comment]
I do not agree, that only inconsistent results are bugs... An Algo not doing what it should do is a bug not just sucks. 

(we develop emergency brake assists. If the brake doesn't brake in an emergency I can hardly say "just algo sucks" :smile: )

But I also understand, the only one who potentially could explain, how far the algo is not following the initial intention, is not loving dt anymore...

**Comment by jenshannoschwalm** [comment]
From what I understand from the code I think that the algo is not good. Haven't read the source  cited original paper though.

How to deal? Don't know. For sure the "AI" is strongly misleading 😎 options:

1. Hide those options as not fulfilling promise, as you should not sell crappy brakes. 
2. Wait for a dev interested in that stuff implementing some other intelligence.

So, still no bug for me 😎 and the issue would be RFC implement something better

**Comment by AxelG-DE** [comment]
> How to deal? Don't know. For sure the "AI" is strongly misleading 😎 options:
> 
>     1. Hide those options as not fulfilling promise, as you should not sell crappy brakes.

If it would be up to me, in that case I vote for "1"

**Comment by ralfbrown** [comment]
I took a look at the code last night, and the name of the method is not at all what it's doing, anyway - pixels which are part of a surface, i.e. have uniform adjacent pixels, are explicitly down-weighted.  So it's actually giving the most weight to edges, where the colors of the neighborhood tend to average much closer to gray.  As a result, the computed shift from the D50 white point is usually miniscule, which is why the CCT gets reported as 5002K +/- a few degrees for most pictures.

So I vote for hiding the option for now, then deciding whether to rewrite the code or remove it entirely in a future release.

**Comment by jenshannoschwalm** [comment]
Some for me.

**Comment by victoryforce** [comment]
+1 vote for hiding
